includes/rest-api/class-sensei-rest-api-internal.php

```diff
@@ -84,15 +84,16 @@ class Sensei_REST_API_Internal {
 	 * Sensei_REST_API_Internal constructor.
 	 */
 	public function __construct() {
-		$remote_data_api = Sensei_Home::instance()->get_remote_data_api();
+		$sensei_home     = Sensei_Home::instance();
+		$remote_data_api = $sensei_home->get_remote_data_api();
 
+		$this->notices_provider     = $sensei_home->get_notices_provider();
```

Comment:
@aaronfc I was wondering what your thoughts were on this. Potentially, we could have `Sensei_Home` manage instantiating all of this and then add a method to Sensei Home that was just `get_rest_api_controller()` that we could fit into this class. What do you think?

Reply:
Sounds good to me 👍

At some point it would be nice to look into adding some dependency injection solution... Until then having everything about Sensei Homr instantiated in the `Sensei_Home` sounds a good compromise :)

Reply:
Cool be7c1e718c88082457bc00be0aefefc1f03cd703